### PR TITLE
F/cloud 1740 skip image

### PR DESCRIPTION
**Pull request by naveen-variant**
# Description

Skip ecr image if it is already generated.

Fixes [#19](https://usxtech.atlassian.net/browse/CLOUD-1740)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

[Demo Flask app](https://github.com/variant-inc/demo-python-flask-variant-api/tree/test/master-postgress)

```yaml
Test Case 1: ECR image should be pushed and existing flow should remain same
  1. Navigate to demo app branch create a commit.
  2. Should trigger run and existing flow should remain same and push image to ECR
  3. Look for log "New image, proceeding to docker push"

Result
  - Expected: See image is pushed with that commit.
  - Actual: 

Test Case 2: Skip image push when re-run
  1. Navigate to actions and pick the run previously created.
  2. Re-run the build, it should trigger the run and should not push image to ECR.
  4. Look for log "Image already pushed. Skipping docker push"

Result
  - Expected: ECR image push should be skipped.
```

- [x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added documentation to test

### CODE REVIEW

**Review by genghis-tuan — COMMENTED** [review]
Looks great. I'll approve once actions-collection is merged and the version for python-action is bumped.

**Comment by zumic96** [comment]
looks good, the test works as expected
https://github.com/variant-inc/demo-python-flask-variant-api/actions/runs/2549338507

**Review by genghis-tuan — APPROVED** [review]

**Review by genghis-tuan — APPROVED** [review]

**Review by jahnavimaddi — APPROVED** [review]
Tested and changes work as expected

**Pull request merged**
